### ISSUE TITLE
PCC of newly trained models is lower

### ISSUE THREAD

**Issue by dbogdano** [mhorlacher/rbpnet#7]
Hello, 

Thanks again for putting together RBPNet. I've been using some of the pre-trained models from the paper as a basis for transfer learning within a number of new models to some success. When trying to train new RBPNet models on eCLIP datasets both used in the paper and not, the Pearson correlation coefficients measured using metrics.PearsonCorrelation are generally not as high as the values described in the paper. In particular, using the encode HepG2 QKI dataset, I'm getting PCCs of between 0.3-0.4, not the 0.830 value noted. Others like the encode K562 MBNL1 dataset trained quite poorly and were not able to identify the signature motifs associated with the RBP. 

I'm wondering if there's a step in the data processing or training that differs between my workflow and the paper's. For eCLIP data processing I'm running the following: 

Get read 2 if paired-end with samtools view, merge replicates with samtools merge and sort with samtools sort 

Generate positive and negative strand bigwig files with bedtools genomecov, with options --strand and -5 or -3 for positive and negative strands respectively, into Encode's bedGraphToBigWig tool.

Use RBPNet's enriched-windows.py script with parameters listed in the paper. 

Use bedtools slop to extend windows to 300bp with equal sized additions on either window side. 

Write the dataspec YAML files with make_dataspec.py

generate the training and validation tfrecords with rbpnet tfrecord, default settings.

training with rbpnet train, using the config.gin listed in the examples directory of this repo, epochs set to 50. 

Most of these steps were taking from instructions in the paper, or this repo. Regarding the results I'm getting, only the poorer QKI PCC is suggestive of something being done differently to achieve the better results in the paper. The new eCLIP datasets I'm training on generally give a PCC of between 0.2-0.3, which is a bit below the average of 0.328 noted in the paper - so it may just be an issue of poor data quality with these new sets. 

Any help would be much appreciated - let me know if I can clarify or expand on any points in the question. 
Thanks
Derek

**Comment by mhorlacher** [comment]
Hi @dbogdano, thanks for your interest in RBPNet and appologies for the late reply, I'm just back from vacation. 

Regarding your first question, it does seem like you've performed all processing steps correctly. However, beware that the cor. coeff. of 0.830 you are referring to is the value for the hold-out sample with the highest number of counts. As reported in the paper, the avg. cor. coef. across all hold-out samples lies between 0.2 and ~0.5, depending on the RBP, so the value you are getting seems to match the results from the paper. Generally, we expect that the cor. coef. will be higher for samples with many counts, not only because there is likely more biological signal, but also because it makes the coef. less prone to noise. 

Regarding lack of motif for MBNL1, this could be due to several reasons, e.g. poor data quality, poor model convergence, or due to post-processing of the feature importance tracks, motif extraction and clustering. In the paper, we extracted fixed-sized sequences of length 5 with highest feature importance from each sequence, which is a relatively naive approach, as the assumption that all RBPs have a fixed-sized motif is violated. To narrow the issue down, you could look at feature importance tracks generated by the model for regions where you know that a binding site should be present and check if the model picks up a motif. 

Regarding QKI performance, are you referring to the eCLIP data from K562? We've only trained on HepG2 in the paper, so in that case it may be due to data quality. Otherwise, could you share the BED file with the final 300bp windows with me? I'd be happy to have a look at it!

**Comment by dbogdano** [comment]
Hi Marc, 

Thanks for the thorough reply, I appreciate it. As noted I've generally been getting Pearson correlation coefficients of around 0.2 -0.5 for the hold out sample test. Though this range includes the HepG2 QKI model, which suggests I might be doing something suboptimal in the preprocessing or training steps. I've been getting similar poorer results for other RBPs of interest that have PCCs noted in the paper (e.g. HepG2-AQR and HepG2-HNRNPC as shown in Fig 2b.). Here's a link to my 300bp window bed files for QKI HepG2, thanks for offering to take a look: https://ucsf.box.com/s/poxb99gochn2m374xg11e00pnkk8fazn
let me know if you have any issues accessing them. Just confirming if the file sizes look correct might be enough to troubleshoot.

The pre-trained QKI model from your paper performs beautifully in my transfer-learning model, so generating some new RBPNet models trained in the same manner would be ideal. 

Regarding motif detection and the MBNL1 model, I've since been able to get the proper YCGY motifs using TF-MoDISco on the integrated gradient values. Just to confirm, were you generating your attribution scores with integrated gradients on the hold-out data only? 

Thanks again for the help, and for putting together RBPNet. Let me know if I can better clarify anything above.

**Comment by dbogdano** [comment]
Hi @mhorlacher , 

Just to quickly follow up on this. It's possible that I'm not calculating these Pearson Correlation Coefficients correctly or in the same manner done in the RBPNet paper. Attached is a notebook with a quick evaluation of the pre-trained HepG2 RBFOX2 model from the paper. For the PCC, I'm taking the mean of the correlations between predictions and counts for 10000 sequences in the hold dataset. At 0.1467. this mean correlation is a bit lower than I'd expect for RBFOX.

Derek
[RBPNet_Model_Evaluation.pdf](https://github.com/user-attachments/files/16968930/RBPNet_Model_Evaluation.pdf)
